src/components/auction/OrderbookTable/helpers.ts

```diff
@@ -0,0 +1,54 @@
+import round from 'lodash.round'
+
+import { PricePoint } from '../../../api/AdditionalServicesApi'
+
+// https://stackoverflow.com/a/55700334
+export const countDecimals = function (num: number) {
+  const text = num.toString()
+  if (text.indexOf('e-') > -1) {
+    const [base, trail] = text.split('e-')
+    const elen = parseInt(trail, 10)
+    const idx = base.indexOf('.')
+    return idx == -1 ? 0 + elen : base.length - idx - 1 + elen
+  }
+  const index = text.indexOf('.')
+  return index == -1 ? 0 : text.length - index - 1
+}
+
+export interface OrderBookTableData {
+  amount: number
+  sum: number
+  mySize: number
+}
+
+export const buildTableData = (bids: PricePoint[], myBids: PricePoint[], granularity: number) => {
+  const rangeVolume = new Map<number, OrderBookTableData>()
+  let cumulativeSum = 0
+
+  for (const bid of bids) {
+    const key = getPriceRangeKey(bid.price, granularity)
+    const currentValue = rangeVolume.get(key) || { amount: 0, sum: cumulativeSum, mySize: 0 }
+    currentValue.amount = currentValue.amount + bid.volume
+    currentValue.sum = currentValue.sum + bid.volume
+    cumulativeSum += bid.volume
+
+    let mySize = 0
+    for (const myBid of myBids) {
+      const priceRange = getPriceRangeKey(myBid.price, granularity)
+      if (priceRange === key) {
+        mySize += round((myBid.volume / currentValue.amount) * 100, 2)
+      }
+    }
+    currentValue.mySize = mySize
+
+    rangeVolume.set(key, currentValue)
+  }
```

Comment:
Here you build a map for a price range going to the data object, and then converting that map to an array. Maybe a comment could help, or a unit test. The inner for calculates `mySize` only using `amount` and data from bids. I'm :thinking:  if that could be extracted and `getPriceRangeKey` mapped over bids.

Anyways for sure the easier to make it clearer is to add a few comments or an ex example.

Reply:
Indeed unit tests would make this easier to follow, I'll add them later.

I couldn't follow your suggestion about mapping `getPriceRangeKey` over bids, could you give an example?

Reply:
Something like

```
const priceRanges = bids.map(bid => getPriceRangeKey(bid, granularity))
```
And then use index to access it.

Reply:
Thanks for the example. In that case I would need another pass of the entire bids array to assign the calculated fields. What I see that can be improved is the myBid price range as it's being calculated for every bid, that can be moved out of the bids loop.